src/main/java/com/mraof/minestuck/entity/DialogueEntity.java

```diff
@@ -0,0 +1,34 @@
+package com.mraof.minestuck.entity;
+
+import com.mraof.minestuck.util.Dialogue;
+import com.mraof.minestuck.util.DialogueManager;
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.world.entity.LivingEntity;
+
+public interface DialogueEntity
+{
+	String DIALOGUE_NBT_TAG = "Dialogue";
+	
+	Dialogue getDialogue();
+	
+	void setDialogue(String location);
```

Comment:
I suggest changing this function to take in a resource location or the dialogue itself, to take away the responsibility of the "string -> resource location" parse or the dialogue lookup away from the `DialogueEntity` implementor + make it possible to set the other things directly.

(Unless you want to update all `setDialogue()` functions immediately, you could then add back an overload to `setDialogue()` that takes a string and then parses it to call the other function)